### PR TITLE
Fix Cassandra listTables denies empty table_name

### LINKED ISSUE #8470: tableName is null in Cassandra connector
I need to perform schema scan for my databases connected to presto. Performing table-level scan looks like:

    SELECT table_name, column_name, data_type
    FROM "some-schema".information_schema.columns
    WHERE table_schema = 'some-schema'
    ORDER BY column_name ASC

It works fine for all SQL databases. When trying to scan Cassandra I'm facing following error:

 > Query ***** failed: tableName is null

It seems to be problem with `CassandraMetadata.java` not supporting empty `table_name` predicate here: https://github.com/prestodb/presto/blob/master/presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java#L188

It seems there's a wrong method used: `prefix.getSchemaName()` where I believe it should be: `if (prefix.getTableName() == null) {`

### PR DESCRIPTION
**Pull request by ebyhr**
Fix #8470

### CODE REVIEW

File: presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java [hunk 1]
```diff
@@ -489,6 +489,17 @@ public void testInsert()
                 "key3", null, 999, null, null, null, "ansi", false, null, null, null, null, null, null, null, null, null, null));
     }
 
+    @Test
+    public void testSelectInformationSchemaColumns()
+    {
+        assertQuery(
+                "SELECT table_schema FROM information_schema.columns WHERE table_schema = 'tpch' LIMIT 1",
```
Comment (kokosing) [inline presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java]: use `ORDER BY` to make query be deterministic

File: presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java [hunk 2]
```diff
@@ -489,6 +489,17 @@ public void testInsert()
...
+                "VALUES 'tpch'");
+        assertQuery(
+                "SELECT table_name FROM information_schema.columns WHERE table_name = 'orders' LIMIT 1",
```
Comment (kokosing) [inline presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java]: same as above

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java [hunk 3]
```diff
@@ -185,7 +185,7 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
 
     private List<SchemaTableName> listTables(ConnectorSession session, SchemaTablePrefix prefix)
     {
-        if (prefix.getSchemaName() == null) {
+        if (prefix.getTableName() == null) {
             return listTables(session, prefix.getSchemaName());
         }
         return ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
```
Comment (kokosing) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java]: What if `schemaName` is `null` here?

Comment (ebyhr) [reply presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java]: At first, I thought when `schemaName` is `null`, `tableName` is also always `null` and ` if (prefix.getTableName() == null)` can handle such case.

SchemaTablePrefix.java
```java
    public SchemaTablePrefix()
    {
        this.schemaName = null;
        this.tableName = null;
    }

    public SchemaTablePrefix(String schemaName)
    {
        this.schemaName = checkNotEmpty(schemaName, "schemaName");
        this.tableName = null;
    }

    public SchemaTablePrefix(String schemaName, String tableName)
    {
        this.schemaName = checkNotEmpty(schemaName, "schemaName");
        this.tableName = checkNotEmpty(tableName, "tableName");
    }
```

File: presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java (hunk 1 above)
Comment (ebyhr) [reply presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java]: FYI, The columns in SELECT and WHERE are same, so I eliminated ORDER BY intentionally.

**Comment by ebyhr** [comment]
@kokosing Thanks for your review! Updated.

**Comment by ebyhr** [comment]
Travis fails with timeout error about hive.
```
...
2018-02-05 08:44:14 INFO: SUCCESS     /    com.facebook.presto.tests.hive.TestHiveCoercion.testHiveCoercionTextFile (Groups: hive_connector, hive_coercion, jdbc) took 28.2 seconds
presto-master_1       | 2018-02-05T08:44:14.281+0545	INFO	query-execution-53	com.facebook.presto.event.query.QueryMonitor	TIMELINE: Query 20180205_025914_00271_ty9x6 :: Transaction:[212f442a-e6ac-4626-a0fb-6371a26a072b] :: elapsed 640ms :: planning 390ms :: scheduling 110ms :: running 126ms :: finishing 14ms :: begin 2018-02-05T08:44:13.625+05:45 :: end 2018-02-05T08:44:14.265+05:45
2018-02-05 08:44:14 INFO: [137 of 415] com.facebook.presto.tests.hive.TestHiveBucketedTables.testIgnorePartitionBucketingIfNotBucketed (Groups: hive_connector)


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

By the way, the commit `9feee01` has succeeded in my folk. 
https://travis-ci.org/ebyhr/presto/builds/337380737

**Comment by kokosing** [comment]
Believe or not, but this travis failure seems to be related with https://github.com/prestodb/presto/issues/9799. Please restart.

**Comment by ebyhr** [comment]
Sorry, I don't have a authorization to restart it in `prestodb` travis.

**Comment by kokosing** [comment]
Restarted

**Comment by ebyhr** [comment]
Thank you.

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**